src/typeconversions.jl

```diff
@@ -60,6 +60,26 @@ function datatype(str::VLen{C}) where {C<:CharType}
     Datatype(API.h5t_vlen_create(type_id))
 end
 
+# Compound types
+
+# These will use finalizers. Close them eagerly to avoid issues.
+datatype(::T) where {T} = Datatype(hdf5_type_id(T), true)
+datatype(::Type{T}) where {T} = Datatype(hdf5_type_id(T), true)
```

Comment:
Isn't this definition redundant with the method right above it?

Reply:
No. The first works on instances of a type. The second works on the type itself.

Reply:
We don't seem to need this definition (on L67)